api/src/paths/project/{projectId}/treatments/year/{year}/delete.ts

```diff
@@ -0,0 +1,117 @@
+import { RequestHandler } from 'express';
+import { Operation } from 'express-openapi';
+import { PROJECT_ROLE } from '../../../../../../constants/roles';
+import { getDBConnection } from '../../../../../../database/db';
+import { HTTP400 } from '../../../../../../errors/custom-error';
+import { authorizeRequestHandler } from '../../../../../../request-handlers/security/authorization';
+import { TreatmentService } from '../../../../../../services/treatment-service';
+import { getLogger } from '../../../../../../utils/logger';
+
+const defaultLog = getLogger('/api/project/{projectId}/treatments/{year}/delete');
+
+export const DELETE: Operation = [
+  authorizeRequestHandler((req) => {
+    return {
+      and: [
+        {
+          validProjectRoles: [PROJECT_ROLE.PROJECT_LEAD, PROJECT_ROLE.PROJECT_EDITOR],
+          projectId: Number(req.params.projectId),
+          year: Number(req.params.year),
```

Comment:
Dont need to provide the year here, this whole object/function is just for user authorization.  (The projectId is relevant because we use it to determine your project role)